### ISSUE TITLE
[Suggestion] Migration to Javascript Using NodeJS

### ISSUE THREAD

**Issue by MuhsinunCool** [NamelessMC/Nameless#873]
* What kind of issue is it (Question, Bug, etc.)? Suggestion

* In which version of NamelessMC did it occur (1.x.x)? 2.x (3.x?)

* How would you describe the issue?
My suggestion is for you guys to migrate to JavaScript using NodeJS instead of PHP. This would obviously require a tremendous amount of effort and determination, however. As such, it may be in the best interest of the developers to possibly overtake this massive overhaul in the distant future. Perhaps 3.x?
From my understanding, NodeJS is basically the JavaScript implementation of PHP. A colleague of mine tells me of how PHP is an outdated language to use for this kind of stuff. According to him, NodeJS will fulfill all the duties of PHP and allow the website to be run on hosts that don't allow PHP/don't have the latest version of PHP. This would be extremely helpful for users who may not be able to afford a fully-fleshed web host or just want to test the forum software out.
If I am not mistaken, using NodeJS will allow the forum software to be run on GitHub Pages, no?

EDIT: Can NodeJS sites be compiled to become static sites? Or is this whole suggestion just stupid?

**Comment by MuhsinunCool** [comment]
Apologies if I sound insanely stupid for giving this idea.

**Comment by samerton** [comment]
Unfortunately it's not as simple as that, web hosts need to specifically support NodeJS, and as Node is still fairly new it's not possible to deploy a NodeJS application on many web hosts. 
It's not something which you can simply upload to a web host (including static hosts such as GitHub Pages), as it needs to bind to certain ports and there are still things you need to do server-side configuration wise, and even then a database would be required (although flatfile like SQLite could always be used).

PHP is more of a universal web software, and is supported by the majority of web hosts. Nameless v2 requires version 5.4+, which itself is over 5 years old so web hosts should support it. I created Nameless in PHP for this reason, and it's also something many devs are familiar with (as it's based on the C language, the same as Java, C++ and so on) so people can jump in and make any modifications easily.

PHP isn't outdated, and whilst there are newer languages out there which may seem more attractive, they're still a lot newer and not as widely supported.

Also, making the switch would be a huge undertaking, and it's simply not feasible at this moment in time I'm afraid. In the future though, who knows?

tl;dr: unfortunately this isn't possible as NodeJS isn't as widely supported (currently) on web hosts as PHP, and PHP is perfectly suitable for the project.

**Comment by MuhsinunCool** [comment]
Understood. Thank you for your time. :)
